### DISCUSSION TITLE
Screen configuration widget confusion

### DISCUSSION THREAD

**Discussion by sweeoon** [osmandapp/OsmAnd#14735] [General]
When configuring widgets on a screen panel, there is an Edit function. I use this to add or remove widgets from a panel, and also to or rearrange the order of widgets displayed. It is easy to use and intuitive.

When in page view of a screen panel, if a "Distance to destination" widget is clicked, a selection box opens showing the "Distance to destination" checkbox **not** ticked for this instance! Why so confusing? This happens with all widgets which are in groups.

I understand "Distance to destination" is part of a group called "Navigation points". Choosing a widget from the group should only be done in the "Available widgets" section **before** it's added to the screen panel. If more widgets from the same group is required, do that there. Once a widget from a group is added to a screen panel, please don't show the group selection box (again) when it's clicked, with an "Add" button? I've accidentaly added more instances of a widget when I click on the "Add" button thinking it's the same as an "OK" button.

The "Available widgets" section should be removed from a screen panel page. It should only be found in the "Edit" dialog box. This would also de-clutter the screen panel's page.

OsmAnd has a habit of duplicationg settings/selections and cluttering up pages unnecessarily. Please be kind to users and stop this practice.

**Comment by vshcherb** [comment]
We're refactoring that part already, so on click you will be able to configure and delete widget and not to add it again. Thanks for feedback.
